public/app/plugins/datasource/cloudwatch/components/ConfigEditor.tsx

```diff
@@ -43,6 +43,22 @@ export const ConfigEditor: FC<Props> = (props: Props) => {
         </InlineField>
       </ConnectionConfig>
 
+      <h3 className="page-heading">CloudWatch Logs</h3>
+      <div className="gf-form-group">
+        <InlineField
+          label="Timeout"
+          labelWidth={28}
+          tooltip="Custom timout for CloudWatch Logs insights queries which have max concurrency limits. Default is 15 minutes. You can use interval syntax, ie 30s, 5m, 2000ms"
+        >
+          <Input
+            width={60}
+            placeholder="15m"
+            value={options.jsonData.logsTimeout || ''}
```

Comment:
Think you need to validate the input here. Either check on change or test against regex on Input.valid prop